src/components/company/Company.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import { useQuery } from 'react-query';
+import { useParams } from 'react-router-dom';
+import { AxiosError } from 'axios';
+import UpdateCompany from './UpdateCompany';
+import { companies } from '../../API/requests';
+import DeleteCompany from './DeleteCompany';
+
+function Company() {
+  const { id }: { id: string } = useParams();
+
+  const [company, setCompany] = useState<Company | null>(null);
+
+  const { isLoading, error } = useQuery<Company, AxiosError>(['company', id], () => companies.getOne(id), {
+    onSuccess: (data) => {
+      console.log(company);
+      setCompany(data);
+    },
+    staleTime: Infinity,
+  });
+  console.log(company);
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (error) {
+    return (
+      <p>
+        An error has occurred: {error.message}. Code: {error.code}
+      </p>
```

Comment:
Maybe create an error component that you can use across all the application (but it will be another branch ;) )